Build failing when trying to add the plugin on an Android module

I'm following the quickstart guide step by step trying to converty my annotation processor from kapt to ksp. 
I'm doing this in a new project so there are no conflicts between kapt and ksp
 
When I try to implement the plugin I receive the following exception:

`Extension of type 'CommonExtension' does not exist. Currently registered extension types: [ExtraPropertiesExtension, KspExtension]
`

I don't really know if it's important but I'm trying to implement it in an Android module. 

These are my plugins in the android module.

`plugins {
    id("com.google.devtools.ksp") version "1.4.30-1.0.0-alpha02"
    id("com.android.application")
    kotlin("android")
}`


If any information is needed I'll be happy to provide and the entire project in its current state is available here.
https://github.com/gilgoldzweig/Gencycler/tree/v3

Would you mind to try the solution in #305? Please keep this issue open even if that works.

It might have something to do with how the plugins are applied. The exception is thrown through [this line](https://github.com/google/ksp/blob/master/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/AndroidPluginIntegration.kt#L60).

I'm still investigating but apparently get stuck. + @yigit @gavra0 any thoughts?

```
Caused by: org.gradle.api.UnknownDomainObjectException: Extension of type 'CommonExtension' does not exist. Currently registered extension types: [ExtraPropertiesExtension, KspExtension]
        at org.gradle.internal.extensibility.ExtensionsStorage.getHolderByType(ExtensionsStorage.java:88)
        at org.gradle.internal.extensibility.ExtensionsStorage.configureExtension(ExtensionsStorage.java:70)
        at org.gradle.internal.extensibility.DefaultConvention.configure(DefaultConvention.java:189)
        at org.gradle.internal.extensibility.DefaultConvention.configure(DefaultConvention.java:184)
        at com.google.devtools.ksp.gradle.AndroidPluginIntegration.decorateAndroidExtension(AndroidPluginIntegration.kt:60)
        at com.google.devtools.ksp.gradle.AndroidPluginIntegration.access$decorateAndroidExtension(AndroidPluginIntegration.kt:37)
        at com.google.devtools.ksp.gradle.AndroidPluginIntegration$applyIfAndroidProject$1.execute(AndroidPluginIntegration.kt:44)
        at com.google.devtools.ksp.gradle.AndroidPluginIntegration$applyIfAndroidProject$1.execute(AndroidPluginIntegration.kt:37)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:248)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:245)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:100)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:95)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
        ...
```

--- a/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/AndroidPluginIntegration.kt
+++ b/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/AndroidPluginIntegration.kt
@@ -18,13 +18,14 @@ package com.google.devtools.ksp.gradle
 
 import com.android.build.api.dsl.AndroidSourceSet
 import com.android.build.api.dsl.CommonExtension
+import com.android.build.gradle.BaseExtension
 import com.google.devtools.ksp.gradle.KspGradleSubplugin.Companion.KSP_MAIN_CONFIGURATION_NAME
 import org.gradle.api.Project
 import org.gradle.api.tasks.SourceSet
 import org.gradle.api.tasks.TaskProvider
 import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinJvmAndroidCompilation
 import java.io.File
-import java.util.*
+import java.util.Locale
 
 /**
  * This helper class handles communication with the android plugin.
@@ -38,10 +39,14 @@ class AndroidPluginIntegration(
     private val kspGradleSubplugin: KspGradleSubplugin
 ) {
 
+    private val agpPluginIds = listOf("com.android.application", "com.android.library", "com.android.dynamic-feature")
+
     fun applyIfAndroidProject(project: Project) {
-        project.pluginManager.withPlugin("com.android.base") {
-            // for android apps, we need a configuration per source set
-            decorateAndroidExtension(project)
+        agpPluginIds.forEach { agpPluginId ->
+            project.pluginManager.withPlugin(agpPluginId) {
+                // for android apps, we need a configuration per source set
+                decorateAndroidExtension(project)
+            }
         }
     }
 
@@ -56,12 +61,16 @@ class AndroidPluginIntegration(
         }
 
     private fun decorateAndroidExtension(project: Project) {
+        val sourceSets = when (val androidExt = project.extensions.getByName("android")) {
+            is BaseExtension -> androidExt.sourceSets
+            is CommonExtension<*, *, *, *, *, *, *, *> -> androidExt.sourceSets
+            else -> throw RuntimeException("Unsupported Android Gradle plugin version.")
+        }
+
         @Suppress("UnstableApiUsage")
-        project.extensions.configure(CommonExtension::class.java) { commonAndroidExtension ->
-            kspGradleSubplugin.run {
-                commonAndroidExtension.sourceSets.createKspConfigurations(project) { androidSourceSet ->
-                    listOf(androidSourceSet.kspConfigurationName)
-                }
+        kspGradleSubplugin.run {
+            sourceSets.createKspConfigurations(project) { androidSourceSet ->
+                listOf(androidSourceSet.kspConfigurationName)
             }
         }
     }

